src/Form/Form.js

```diff
@@ -895,10 +896,18 @@ class Form {
 
         // After autofill we check if form values match the data provided…
         const formValues = this.getValuesReadyForStorage();
+        const hasNoCredentialsData = !formValues.credentials?.username && !formValues.credentials?.password;
+        const hasOnlyEmail = formValues.identities && Object.keys(formValues.identities).length === 1 && formValues.identities.emailAddress;
+
+        const hasOnlyOneCredentialOrEmail =
+            Boolean(formValues.credentials?.username) !== Boolean(formValues.credentials?.password) ||
+            (hasOnlyEmail && hasNoCredentialsData);
```

Comment:
I'm a little unclear why we'd need this extra logic. In theory, here we're checking whether the data we autofilled is _the only data_ in the form (`areAllFormValuesKnown`). Why do we need these extra checks here? It _feels like_ it should just work 🤔.

Reply:
In the case of single field forms, `areAllFormValuesKnown` is always true, leading to `shouldPromptToStoreData` to always be `false`.
This causes `storeFormData` call to be not triggered, in `postSubmit`.

Maybe we can do the change inside of `areAllFormValuesKnown` and that way it's probably clearer.

Reply:
This is only problematic if you have a `username-only` entry stored, and you autofilled that and later we want to use that to reconcile the password.

(Sorry I keep forgetting these awkward cases :D).